core/src/page_id.rs

```diff
@@ -0,0 +1,170 @@
+//! TODO: add docs
+
+use crate::{page::DEPTH, trie::KeyPath};
+use bitvec::{prelude::*, slice::ChunksExact};
+
+pub type PageId = [u8; 32];
+
+pub const ROOT_PAGE_ID: [u8; 32] = [0; 32];
+
+const MAX_CHILD_INDEX: u8 = (1 << DEPTH) - 1;
+
+pub fn child_page_id(mut page_id: PageId, child_index: u8) -> PageId {
+    // TODO: proper error handling
+    assert!(child_index & 0b11000000 == 0);
+
+    let page_id_bits = page_id.view_bits_mut::<Msb0>();
+
+    if child_index ^ MAX_CHILD_INDEX == 0 {
```

Comment:
why not just use `==`? seems unnecessarily fancy

Reply:
Totally true, I started thinking about using a mask and I went in that direction but yes,`==` does the same thing and it is cleaner